### PR TITLE
awatcher: init at 0.2.7

### PR DESCRIPTION
**Pull request by tsandrini**
## Description of changes

[awatcher](https://github.com/2e3s/awatcher) is an alternative activity and idle watcher for [ActivityWatch](https://github.com/ActivityWatch/activitywatch). This can be useful for some folks in certain occasions where `aw-watcher-window`  - https://github.com/NixOS/nixpkgs/blob/755b915a158c9d588f08e9b08da9f7f3422070cc/pkgs/applications/office/activitywatch/default.nix#L58

doesn't really work, for example when using KWin on wayland + other neat features.

---

**Note**: Since `awatcher` doesn't belong to the ActivityWatch org I thought it should probably be put into it's own separate `pkgs/by-name/aw/awatcher/` dir.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tsandrini** [comment]
Output of `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 321249"`:

```shell
1 package added:
awatcher (init at 0.2.7)
```

File: pkgs/by-name/aw/awatcher/package.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  openssl,
+  ...
```
Comment (applePrincess) [inline pkgs/by-name/aw/awatcher/package.nix:7]: Delete this line

File: pkgs/by-name/aw/awatcher/package.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  # NOTE needed due to Cargo.lock containing git dependencies
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "aw-client-rust-0.1.0" = "sha256-fCjVfmjrwMSa8MFgnC8n5jPzdaqSmNNdMRaYHNbs8Bo=";
+    };
+  };
+
+  meta = with lib; {
+    description = "Awatcher is a window activity and idle watcher for ActivityWatcher with an optional tray and UI for statistics.";
```
Comment (applePrincess) [inline pkgs/by-name/aw/awatcher/package.nix:33]:
```suggestion
    description = "A window activity and idle watcher for ActivityWatcher with an optional tray and UI for statistics";
```

File: pkgs/by-name/aw/awatcher/package.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    homepage = "https://github.com/2e3s/awatcher";
+    changelog = "https://github.com/2e3s/awatcher/releases/tag/${version}";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ tsandrini ];
+    platforms = platforms.linux;
+    mainProgram = pname;
```
Comment (applePrincess) [inline pkgs/by-name/aw/awatcher/package.nix:39]:
```suggestion
    mainProgram = "awatcher";
```

**Review by applePrincess — CHANGES_REQUESTED** [review]

**Comment by tsandrini** [comment]
@applePrincess Thanks for the feedback! I updated the code accordingly :)

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 321249` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>awatcher</li>
  </ul>
</details>

**Review by applePrincess — APPROVED** [review]

**Comment by Guekka** [comment]
This PR seems to be a duplicate of #287858

**Comment by tsandrini** [comment]
@Guekka Oh, my bad, I was sure that I looked through issues/PR for an existing one but probably just blind & stupid. Should I close this then or? Not sure what it is about the v3.0.0, the current master beta is v3.0.0. but they haven't yet released a v3.0.0 version hence why I opted for the latest one. Might just have been an accidental release, not sure.

**Comment by Guekka** [comment]
> @Guekka Oh, my bad, I was sure that I looked through issues/PR for an existing one but probably just blind & stupid. Should I close this then or? Not sure what it is about the v3.0.0, the current master beta is v3.0.0. but they haven't yet released a v3.0.0 version hence why I opted for the latest one. Might have been just an accidental release, not sure.

It happens, don't worry. Actually, your PR is even the third one #315864 :p 
Sorry for your duplicated work. I guess we can keep both PRs open until we get more information, maybe @aikooo7 could explain?

**Comment by tsandrini** [comment]
Sooo, out of curiosity I checked my local awatcher repo and I found a locally cached `v3.0.0` tag so maybe they really created it and deleted it? Possibly to bring more features? Not really sure, however #287858 builds without any issues for me so ig okay?

**Comment by aikooo7** [comment]
> > @Guekka Oh, my bad, I was sure that I looked through issues/PR for an existing one but probably just blind & stupid. Should I close this then or? Not sure what it is about the v3.0.0, the current master beta is v3.0.0. but they haven't yet released a v3.0.0 version hence why I opted for the latest one. Might have been just an accidental release, not sure.
> 
> It happens, don't worry. Actually, your PR is even the third one #315864 :p Sorry for your duplicated work. I guess we can keep both PRs open until we get more information, maybe @aikooo7 could explain?

At the time 3.0.0 existed, if it doesn't anymore, I will change my pr soon

**Comment by aikooo7** [comment]
> > > @Guekka Oh, my bad, I was sure that I looked through issues/PR for an existing one but probably just blind & stupid. Should I close this then or? Not sure what it is about the v3.0.0, the current master beta is v3.0.0. but they haven't yet released a v3.0.0 version hence why I opted for the latest one. Might have been just an accidental release, not sure.
> > 
> > 
> > It happens, don't worry. Actually, your PR is even the third one #315864 :p Sorry for your duplicated work. I guess we can keep both PRs open until we get more information, maybe @aikooo7 could explain?
> 
> At the time 3.0.0 existed, if it doesn't anymore, I will change my pr soon

Should be good to go now :)

**Comment by Guekka** [comment]
Alright, let's close this PR since the other one is older. Thank you for your work

**Pull request closed without merging**